README.md

```diff
@@ -1,213 +1,158 @@
+methods on a live instance.
 
+Interface functions may be suspending. Internally Zipline implements `setTimeout()` to make
+asynchronous code work as it's supposed to in Kotlin/JS.
 
-## Building
+Zipline also supports `Flow<T>` as a parameter or return type. This makes it easy to build reactive
+systems.
 
-### For Android
 
-```
-$ ./gradlew :quickjs:android:build
-```
+### Fast
 
-The build system will automatically cross-compile the native library to all applicable ABIs and
-create the fat `.aar` in `quickjs/android/build/outputs/aar/`.
+One potential bottleneck of embedding JavaScript is waiting for the engine to compile the input
+source code. Zipline precompiles JavaScript into efficient QuickJS bytecode to eliminate this
+performance penalty.
 
-This will not run the tests. You must have a working device or emulator connected at which point
-you can explicitly run them.
+Another bottleneck is waiting for code to download. Zipline addresses this with support for modular
+applications. Each input module (Like Kotlin's standard, serialization, and coroutines libraries)
+is downloaded concurrently. Each downloaded module is cached. Modules can also be embedded with the
+host application to avoid any downloads if the network is unreachable. If your application module
+changes more frequently than your libraries, users only download what's changed.
 
-```
-$ ./gradlew :quickjs:android:connectedCheck
-```
+If you run into performance problems in the QuickJS runtime, Zipline includes a sampling profiler.
+You can use this to get a breakdown of how your application spends its CPU time.
 
-### For the JVM
 
-First, build the native library for your host OS:
-```
-$ cmake -S zipline/src/jvmMain/ -B build/jni/ -DQUICKJS_VERSION="$(cat zipline/native/quickjs/VERSION)"
-$ cmake --build build/jni/ --verbose
-```
+### Developer-Friendly
 
-Next, copy the resulting binary into the resources of the JVM project:
-```
-$ mkdir -p zipline/src/jvmMain/resources/
-$ cp -v build/jni/libquickjs.* zipline/src/jvmMain/resources/
-```
-
-Finally, build the platform-specific `.jar` and run the tests:
-```
-$ ./gradlew :quickjs:jvm:build
-```
+Zipline implements `console.log` by forwarding messages to the host platform. It uses
+`android.util.Log` on Android, `java.util.logging` on JVM, and `stdout` on Kotlin/Native.
 
-The `.jar` will be available in `quickjs/jvm/build/libs/`.
+Zipline integrates Kotlin source maps into QuickJS bytecode. If your process crashes, the stacktrace
+will print `.kt` files and line numbers. Even though there's JavaScript underneath, developers don't
+need to interface with `.js` files.
 
+After using a bridged interface it must be closed so the peer object can be garbage collected. This
+is difficult to get right, so Zipline borrows ideas from [LeakCanary] and aggressively detects
+when a `close()` call is missed.
 
-## License
+### License
 
-    Copyright 2015 Square, Inc.
+    Copyright 2022 Block, Inc.
```

Comment:
We've never updated these in the past

Reply:
Yeah, that’s an artifact of me starting from scratch, gathering stuff from a few projects. Fixed.